clients/spark/core/src/test/resources/parser-test/sst_files_generator.go

```diff
@@ -0,0 +1,286 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"math/rand"
+	"os"
+	"sort"
+
+	fuzz "github.com/google/gofuzz"
+
+	"github.com/cockroachdb/pebble/sstable"
+	nanoid "github.com/matoous/go-nanoid/v2"
+)
+
+const (
+	DefaultKeySizeBytes   = 100
+	DefaultValueSizeBytes = 5
+	MbToBytes             = 1024 * 1024
+	KbToBytes             = 1024
+	FuzzerSeed            = 50
+	// DefaultCommittedPermanentMaxRangeSizeMb - the max file size that can be written by graveler (pkg/config/config.go)
+	DefaultCommittedPermanentMaxRangeSizeMb = 20
+)
+
+func getDefaultUserProperties() map[string]string {
+	return map[string]string{
+		"user_prop_1": "val1",
+		"user_prop_2": "val2",
+	}
+}
+
+type Entry struct {
+	Key   string
+	Value string
+}
+
+func main() {
+	generateTwoLevelIdxSst()
+	fuzzSstContents()
+	fuzzWriterOptions()
+	sstsWithUnsupportedWriterOptions()
+	generateLargeSsts()
+}
+
+func sstsWithUnsupportedWriterOptions() {
+	sizeBytes := 10 * KbToBytes
+	keys := generateSortedSlice(sizeBytes)
+
+	// Checksum specifies which checksum to use. lakeFS supports parsing sstables with crc checksum.
+	// ChecksumTypeXXHash and ChecksumTypeNone are unsupported and therefore not tested.
```

Comment:
Do we explicitly write our SSTables with these options?  If not, should we test (and fix...) these other options?